mongoimport/mongoimport.go

```diff
@@ -154,17 +161,30 @@ func (imp *MongoImport) ValidateSettings(args []string) error {
 		}
 	}
 
+	if imp.IngestOptions.Upsert == true {
+		imp.IngestOptions.Mode = MODE_UPSERT
+	}
+	if imp.IngestOptions.Mode == "" {
+		imp.IngestOptions.Mode = MODE_INSERT
+	} else if !(imp.IngestOptions.Mode == MODE_INSERT ||
+		imp.IngestOptions.Mode == MODE_UPSERT ||
+		imp.IngestOptions.Mode == MODE_MERGE) {
+		return fmt.Errorf("invalid --mode argument: %v", imp.IngestOptions.Mode)
+	}
+
 	if imp.IngestOptions.UpsertFields != "" {
-		imp.IngestOptions.Upsert = true
+		if imp.IngestOptions.Mode == MODE_INSERT {
+			imp.IngestOptions.Mode = MODE_UPSERT
```

Comment:
--upsertFields can be used alone before. are you sure to throw error here?

if throws error here, someone should encounter a failure if they use --upsertFields alone.